csharp/ql/lib/semmle/code/csharp/Element.qll

```diff
@@ -135,23 +118,12 @@ class NamedElement extends Element, @named_element {
    * Unbound generic types, such as `IList<T>`, are represented as
    * ``System.Collections.Generic.IList`1``.
    */
-  cached
-  final string getFullyQualifiedName() {
+  deprecated final string getFullyQualifiedName() {
```

Comment:
Actually, instead of deprecating this predicate, we could just uncache it, and add `bindingset[this]` and `pragma[inline_late]`.

Reply:
I also considered that, but I wan't to prevent future me from using the predicate by mistake somewhere in the production code. Maybe it should be a bit developer hostile to construct the fully qualified name to avoid the temptation of using it? :-)

Reply:
We could also call it `getFullyQualifiedNameForDebug`, and mention in the QL doc that it should only be used for debugging.

Reply:
Yes, that sounds like a good idea. I will add such a predicate with some comments on the footgun of using it in production.